Rlabkey/R/labkey.webdav.R

```diff
@@ -0,0 +1,230 @@
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, and remoteFilePath"));
+    }
+    
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath);
+
+    return(paste(baseUrl, "_webdav", folderPath, fileSet, "/", remoteFilePath, sep=""))
+}
+
+labkey.webdav.pathExists <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    if (missing(baseUrl) || is.null(baseUrl) || missing(folderPath) || missing(remoteFilePath)) {
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, and remoteFilePath"));
+    }
+    
+    ret <- labkey.webdav.listDir(baseUrl, folderPath, fileSet, remoteFilePath)
+    return(is.null(ret$exception))
+}
+
+labkey.webdav.listDir <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    url <- labkey.webdav._validateRemoteUrl(baseUrl = baseUrl, folderPath = folderPath, fileSet = fileSet, remoteFilePath = remoteFilePath)
+    url <- paste0(url, "?method=JSON")
+    content <- labkey.post(url, pbody = "", responseType = "text/plain; charset=utf-8")
+
+    # The intent of this is to mask some of the properties only intended for rendering the file browser UI (like icon)
+    ret <- fromJSON(content, simplifyVector=FALSE, simplifyDataFrame=FALSE)
+    colNames <- c("id", "href", "text", "creationdate", "createdby", "lastmodified", "contentlength", "size", "isdirectory")
+    ret[["files"]] <- lapply(ret[["files"]], function(l){
+        idx <- match("collection", names(l))
+        if (!is.na(idx)){
+            names(l)[idx] <- "isdirectory"
+        } else {
+            l$isdirectory <- FALSE
+        }
+
+        l <- l[colNames]
+        names(l) <- colNames
+        return(l)
+    })
+
+    return(ret)
+}
+
+labkey.webdav.delete <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    url <- labkey.webdav._validateRemoteUrl(baseUrl = baseUrl, folderPath = folderPath, fileSet = fileSet, remoteFilePath = remoteFilePath)
+    url <- paste0(url, "?method=DELETE")
+
+    labkey.post(url, pbody = "", responseType = "text/plain; charset=utf-8")
+}
+
+labkey.webdav.mkDirs <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    if (missing(baseUrl) || is.null(baseUrl) || missing(folderPath) || missing(remoteFilePath)){
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, and remoteFilePath"))
+    }
+
+    remoteFilePaths <- strsplit(remoteFilePath, "/")[[1]]
+    toCreate <- ""
+    for (folderName in remoteFilePaths) {
+        toCreate <- paste0(toCreate, folderName, "/")
+        if (!labkey.webdav.pathExists(baseUrl, folderPath, fileSet, toCreate)) {
+            if (!labkey.webdav.mkDir(baseUrl, folderPath, fileSet, toCreate)){
```

Comment:
this is not working for me since fileSet and toCreate order doesn't match defined arguments and toCreate is not a matched param

Reply:
yes, i should have seen that.  the result of switching argument order and not using named arguments.  i'm gonna switch everything to use named args within labkey.webdav